theano/gof/tests/test_graph.py

```diff
@@ -361,7 +361,7 @@ def test_constant(self):
         r3 = tensor.constant(1.6)
         # The cache still create a new object that we don't return.
         # This is why we must increase by 2 and not 1.
-        assert r3.auto_name == "auto_" + str(autoname_id + 2)
+        assert r3.auto_name == "auto_" + str(autoname_id + 1)
```

Comment:
done.

Mostly, the old code was creating a tmp variable in tensor.constant() and not returining it when it is in the cache. Now we only create a variable if it is not in the cache. So now a call to constant() don't always increment the number.